src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -349,98 +340,132 @@ public static SchemaDefinition Create(Type userType, Direction direction = Direc
+        {
+            switch (memberInfo)
             {
-                // Clause to handle the field that may be used to expose the cursor channel.
-                // This field does not need a column.
-                // REVIEW: maybe validate the channel attribute now, instead
-                // of later at cursor creation.
-                switch (memberInfo)
-                {
-                    case FieldInfo fieldInfo:
-                        if (fieldInfo.FieldType == typeof(IChannel))
-                            continue;
+                case FieldInfo fieldInfo:
+                    if (fieldInfo.FieldType == typeof(IChannel))
+                        return false;
 
-                        // Const fields do not need to be mapped.
-                        if (fieldInfo.IsLiteral)
-                            continue;
+                    // Const fields do not need to be mapped.
+                    if (fieldInfo.IsLiteral)
+                        return false;
 
-                        break;
+                    break;
 
-                    case PropertyInfo propertyInfo:
-                        if (propertyInfo.PropertyType == typeof(IChannel))
-                            continue;
-                        break;
+                case PropertyInfo propertyInfo:
+                    if (propertyInfo.PropertyType == typeof(IChannel))
+                        return false;
+                    break;
 
-                    default:
-                        Contracts.Assert(false);
-                        throw Contracts.ExceptNotSupp("Expected a FieldInfo or a PropertyInfo");
-                }
+                default:
+                    Contracts.Assert(false);
+                    throw Contracts.ExceptNotSupp("Expected a FieldInfo or a PropertyInfo");
+            }
+
+            if (memberInfo.GetCustomAttribute<NoColumnAttribute>() != null)
+                return false;
 
-                if (memberInfo.GetCustomAttribute<NoColumnAttribute>() != null)
-                    continue;
+            return true;
+        }
 
-                var customAttributes = memberInfo.GetCustomAttributes();
-                var customTypeAttributes = customAttributes.Where(x => x is DataViewTypeAttribute);
-                if (customTypeAttributes.Count() > 1)
-                    throw Contracts.ExceptParam(nameof(userType), "Member {0} cannot be marked with multiple attributes, {1}, derived from {2}.",
-                        memberInfo.Name, customTypeAttributes, typeof(DataViewTypeAttribute));
-                else if (customTypeAttributes.Count() == 1)
-                {
-                    var customTypeAttribute = (DataViewTypeAttribute)customTypeAttributes.First();
-                    customTypeAttribute.Register();
-                }
+        public static bool ValidateMemberInfo(MemberInfo memberInfo, Type userType, HashSet<string> colNames, out string name, out IEnumerable<Attribute> customAttributes)
```

Comment:
I think it would be better to rename this to `GetNameAndCustomAttributes` or something like that.